Build throws error after docusaurus 2.4.0 upgrade

Build log:

[INFO] [en] Creating an optimized production build...
ℹ️ Compiling Client
ℹ️ Compiling Server
✔️ Client: Compiled successfully in 4.84m
✔️ Server: Compiled with some errors in 5.22m

[ERROR] Docusaurus server-side rendering could not render static page with path /search.

TypeError: (0 , _docusaurus_theme_common_internal__WEBPACK_IMPORTED_MODULE_15__.useSearchPage) is not a function

[INFO] Docusaurus version: 2.4.0
Node version: v18.12.1

--- a/scripts/generate_documents.sh
+++ b/scripts/generate_documents.sh
@@ -89,9 +89,9 @@ Install_Docusaurus() {
   # semver[2] is minor version
   # semver[3] is patch
   if [ -z "${semver[1]}" ]; then
-    if ! npx create-docusaurus@2.3.1 "${ARG_BUILD_DIR:?}/devdocs" classic --typescript;
+    if ! npx create-docusaurus@2.4 "${ARG_BUILD_DIR:?}/devdocs" classic --typescript;
     then
-      >&2 echo "npx create-docusaurus@2.3.1 failed exiting"
+      >&2 echo "npx create-docusaurus@2.4 failed exiting"
       exit 1
     fi
     # add another module exit if we can't get into the directory

--- a/web/docusaurus/src/theme/SearchBar/index.js
+++ b/web/docusaurus/src/theme/SearchBar/index.js
@@ -5,8 +5,10 @@ import {useHistory} from '@docusaurus/router';
 import {useBaseUrlUtils} from '@docusaurus/useBaseUrl';
 import Link from '@docusaurus/Link';
 import Head from '@docusaurus/Head';
-import {isRegexpStringMatch} from '@docusaurus/theme-common';
-import {useSearchPage} from '@docusaurus/theme-common/internal';
+import {
+  isRegexpStringMatch,
+  useSearchLinkCreator,
+} from '@docusaurus/theme-common';
 import {DocSearchButton, useDocSearchKeyboardEvents} from '@docsearch/react';
 import {useAlgoliaContextualFacetFilters} from '@docusaurus/theme-search-algolia/client';
 import Translate from '@docusaurus/Translate';
@@ -19,9 +21,9 @@ function Hit({hit, children}) {
 }
 
 function ResultsFooter({state, onClose}) {
-  const {generateSearchPageLink} = useSearchPage();
+  createSearchLink = useSearchLinkCreator();
   return (
-    <Link to={generateSearchPageLink(state.query)} onClick={onClose}>
+    <Link to={createSearchLink(state.query)} onClick={onClose}>
       <Translate
         id="theme.SearchBar.seeAll"
         values={{count: state.context.nbHits}}>
@@ -139,7 +141,7 @@ function DocSearch({contextualSearch, externalUrlRegex, ...props}) {
     },
     [siteMetadata.docusaurusVersion],
   );
-  
+
   useDocSearchKeyboardEvents({
     isOpen,
     onOpen,

--- a/web/docusaurus/src/theme/SearchPage/index.js
+++ b/web/docusaurus/src/theme/SearchPage/index.js
@@ -14,11 +14,9 @@ import {
   usePluralForm,
   isRegexpStringMatch,
   useEvent,
+  useSearchQueryString,
 } from '@docusaurus/theme-common';
-import {
-  useTitleFormatter,
-  useSearchPage,
-} from '@docusaurus/theme-common/internal';
+import {useTitleFormatter} from '@docusaurus/theme-common/internal';
 import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
 import { useAllDocsData } from '@docusaurus/plugin-content-docs/client';
 import Translate, { translate } from '@docusaurus/Translate';
@@ -227,7 +225,7 @@ function SearchPageContent() {
   } = themeConfig;
   const documentsFoundPlural = useDocumentsFoundPlural();
   const docsSearchVersionsHelpers = useDocsSearchVersionsHelpers();
-  const { searchQuery, setSearchQuery } = useSearchPage();
+  const [searchQuery, setSearchQuery] = useSearchQueryString();
   const initialSearchResultState = {
     items: [],
     query: null,
@@ -463,7 +461,7 @@ function SearchPageContent() {
 
         <div className={styles.containerSearchPage}>
           {/* <div className={styles.searchSidebar}>
-          
+
         </div> */}
 
           <div className={styles.resultsContainer}>

